internal/ci/ci.go

```diff
@@ -81,3 +94,179 @@ func GetProvider() Provider {
 
 	return None
 }
+
+func getCI(provider Provider) CI {
+	var ci CI
+	if reflect.DeepEqual(provider, AppVeyor) {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: os.Getenv("APPVEYOR_URL"),
+			Repo:      os.Getenv("APPVEYOR_REPO_NAME"),
+			RefName:   os.Getenv("APPVEYOR_PULL_REQUEST_HEAD_REPO_BRANCH"),
+			SHA:       os.Getenv("APPVEYOR_REPO_COMMIT"),
+			User:      os.Getenv("APPVEYOR_REPO_COMMIT_AUTHOR"),
+		}
+	}
+	if reflect.DeepEqual(provider, AWS) {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: os.Getenv("CODEBUILD_PUBLIC_BUILD_URL"),
+			Repo:      os.Getenv("CODEBUILD_SOURCE_REPO_URL"),
+			RefName:   os.Getenv("CODEBUILD_SOURCE_VERSION"),
+			SHA:       os.Getenv("CODEBUILD_RESOLVED_SOURCE_VERSION"),
+			User:      os.Getenv("CODEBUILD_WEBHOOK_ACTOR_ACCOUNT_ID"),
+		}
+	}
+	if reflect.DeepEqual(provider, Azure) {
+		ci = CI{
+			Provider:  provider,
+			OriginURL: os.Getenv("Build_BuildUri"),
+			Repo:      os.Getenv("System_PullRequest_SourceRepositoryURI"),
+			RefName:   os.Getenv("Build_SourceBranchName"),
+			SHA:       os.Getenv("Build_SourceVersion"),
+			User:      os.Getenv("Build_Reason"),
```

Comment:
I don't see how `Build_Reason` maps to user? 🤔

Reply:
updated